spec/gate/indexSpec.js

```diff
@@ -89,4 +89,27 @@ describe('UserGate', function() {
       expect(gate.allows('bar@mixmax.com')).toBe(true);
     });
   });
+
+  describe('uniform sample', () => {
+    it('should be deterministic', () => {
+      const gate = new UserGate({ sample: 0.5 });
+      expect(gate._matchesUniformSample('alice@mixmax.com')).toBe(true);
```

Comment:
From what I understand it should be coincidence that `alice@mixmax.com` passes the gate and `ziggy@mixmax.com` fails, since we're hashing their emails (perhaps this is copied from a test where we used the character check?). To guard against changing hashing algorithms in the future, can we have the test do:
```js
const aliceIsAllowed = gate._matchesUniformSample('alice@mixmax.com');
expect(gate._matchesUniformSample('alice@mixmax.com')).toBe(aliceIsAllowed);
```

Reply:
https://github.com/mixmaxhq/user-gate/pull/5/commits/bbaee3d607cd7dc45dd72ebe13de72fb55c7c6d1